src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -170,55 +205,141 @@ public IList<NameValueHeaderValue> Parameters
         /// </summary>
-        public bool MatchesAllTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"json"</c>.
+        /// </example>
+        public StringSegment SubTypeSuffix
         {
             get
             {
-                return MediaType.Equals("*/*", StringComparison.Ordinal);
+                var subTypeSuffixLength = _mediaType.LastIndexOf(PlusCharacter);
+                if (subTypeSuffixLength == -1)
+                {
+                    return default(StringSegment);
+                }
+                return _mediaType.Subsegment(subTypeSuffixLength + 1);
             }
         }
 
+
         /// <summary>
-        /// SubType = "*"
+        /// Get a <see cref="IList{T}"/> of facets of the <see cref="MediaTypeHeaderValue"/>. Facets are a
+        /// period separated list of StringSegments in the <see cref="SubTypeWithoutSuffix"/>.
+        /// See <see href="https://tools.ietf.org/html/rfc6838#section-3">The RFC documentation on facets.</see>
         /// </summary>
-        public bool MatchesAllSubTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value:
+        /// <c>{"vnd", "example"}</c>
+        /// </example>
+        public IList<StringSegment> Facets
         {
             get
             {
-                return SubType.Equals("*", StringComparison.Ordinal);
+                return SubTypeWithoutSuffix.Split(new char[] { PeriodCharacter } ).ToList();
             }
         }
 
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all types.
+        /// </summary>
+        public bool MatchesAllTypes => MediaType.Equals(MatchesAllString, StringComparison.Ordinal);
+
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all subtypes.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/json"</c>, this property is <c>false</c>.
+        /// </example>
+        public bool MatchesAllSubTypes => SubType.Equals(WildcardString, StringComparison.Ordinal);
+
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all subtypes, ignoring any structured syntax suffix.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*+json"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, this property is <c>false</c>.
+        /// </example>
+        public bool MatchesAllSubTypesWithoutSuffix =>
```

Comment:
@Tratcher @davidfowl do you object to having this in the HTTP abstractions codebase? I'm not sure any spec says anything about wildcard and 'matching' but it's the common and obvious usage.

Reply:
@Tratcher Yes or no on separating this into its own class? I'd prefer to just keep these as is, staying in line with MVC. However I can be convinced otherwise.

Reply:
Is there a line you can draw between "This is the header type and basic information about it" vs "This is content negotiation"? Which members would be pulled out?

Reply:
I'd say none. I think that these properties provide basic information about the header.

Reply:
IsSubsetOf gets pretty close, but that's an existing method we can't move.